### PR TITLE
[ATL][ATLTEST] Adapt to RATL and improvements

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Adapt ReactOS ATL for RATL.
JIRA issue: N/A
Independent ReactOS ATL (RATL): https://github.com/katahiromz/RATL
## Proposed changes
- Modify `atldef.h` for non-ReactOS environments.
- Update ATL testcase project files (from `v120_xp` to `v140_xp`).
- `#define HAVE_APITEST` and use it.
- `s/NULL/0/`

### CODE REVIEW

File: sdk/lib/atl/atldef.h [hunk 1]
```diff
@@ -7,6 +7,14 @@
 
 #pragma once
 
+#ifndef __REACTOS__
+    #ifdef __cplusplus
+        #include <cstddef>
```
Comment (ThFabba) [inline sdk/lib/atl/atldef.h]: Is this necessary? C++ supports <stddef.h> just fine

Comment (katahiromz) [reply sdk/lib/atl/atldef.h]: Moved to `CAtlFileMapping.cpp`.

File: modules/rostests/CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,8 @@
 
 start_module_group(rostests)
 
+add_definitions(-DHAVE_APITEST)
```
Comment (learn-more) [inline modules/rostests/CMakeLists.txt]: Maybe something like `IS_REACTOS_APITEST` ?

File: sdk/lib/atl/atlbase.h [hunk 3]
```diff
@@ -1074,11 +1074,11 @@ class CRegKey
         ATLASSERT(lpszKeyName);
 
         HKEY hKey = NULL;
-        LONG lRes = ::RegOpenKeyEx(hKeyParent, lpszKeyName, NULL, samDesired, &hKey);
+        LONG lRes = ::RegOpenKeyEx(hKeyParent, lpszKeyName, 0, samDesired, &hKey);
```
Comment (learn-more) [inline sdk/lib/atl/atlbase.h]: Why?

File: sdk/lib/atl/atldef.h [hunk 4]
```diff
@@ -7,6 +7,9 @@
 
 #pragma once
 
+#ifndef __REACTOS__
+    #include <pseh/pseh2.h>
+#endif
```
Comment (learn-more) [inline sdk/lib/atl/atldef.h]: Why is this needed?
`PSEH` is a reactos thing.

File: modules/rostests/CMakeLists.txt (hunk 2 above)
Comment (katahiromz) [reply modules/rostests/CMakeLists.txt]: `HAVE_APITEST` is simple and familiar to CMake / `autoconf` culture.

File: sdk/lib/atl/atlbase.h (hunk 3 above)
Comment (katahiromz) [reply sdk/lib/atl/atlbase.h]: The 3rd argument of `RegOpenKeyEx` is integer type. `NULL` for it makes warnings.

File: sdk/lib/atl/atldef.h (hunk 4 above)
Comment (katahiromz) [reply sdk/lib/atl/atldef.h]: Now `RATL` has `PSEH`.

File: modules/rostests/apitests/atl/CAtlFileMapping.cpp [hunk 5]
```diff
@@ -6,8 +6,11 @@
  */
 
 #include <atlfile.h>
+#ifndef __REACTOS__
+    #include <cstddef>
```
Comment (learn-more) [inline modules/rostests/apitests/atl/CAtlFileMapping.cpp]: Is this file included by normal ATL?
Can we otherwise test if it is already included, instead of hanging this on `__REACTOS__` ?

Comment (katahiromz) [reply modules/rostests/apitests/atl/CAtlFileMapping.cpp]: Below, `offsetof` macro needs this.

Comment (HBelusca) [reply modules/rostests/apitests/atl/CAtlFileMapping.cpp]: I was checking where `offsetof()` was used in ATL library code: https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/lib/atl/atlcom.h;hb=453e5bc2afacf8c9595a50a61714febf21636135#l627
I don't know whether manually defining this macro can be done ?

Comment (learn-more) [reply modules/rostests/apitests/atl/CAtlFileMapping.cpp]: Let's not go there unless we absolutely have to.

Comment (katahiromz) [reply modules/rostests/apitests/atl/CAtlFileMapping.cpp]: OK.
